src/join.rs

```diff
@@ -113,6 +131,104 @@ impl<J: Join> std::iter::Iterator for JoinIter<J> {
     }
 }
 
+/// `JoinParIter` is an `ParallelIterator` over a group of `Storages`.
+#[must_use]
+pub struct JoinParIter<J: Join>(J);
+
+impl<J> ParallelIterator for JoinParIter<J>
+  where
+    J: Join + Send,
+    J::Type: Send,
+    J::Value: Split + Send,
+    J::Mask: Send + Sync,
+{
+    type Item = J::Type;
+
+    fn drive_unindexed<C>(self, consumer: C) -> C::Result
+        where C: UnindexedConsumer<Self::Item>
+    {
+        let (keys, values) = self.0.open();
+        let producer = BitProducer((&keys).iter());
+        let values = Arc::new(UnsafeCell::new(values));
+        bridge_unindexed(JoinProducer::<J>::new(producer, values), consumer)
+    }
+}
+
+struct JoinProducer<'a, 'b, J>
+  where
+    J: Join + Send,
+    J::Type: Send,
+    J::Value: 'a + Send,
+    J::Mask: 'b + Send + Sync,
+{
+    keys: BitProducer<'b, J::Mask>,
+    values: Arc<UnsafeCell<J::Value>>,
```

Comment:
I could actually combine the two different lifetimes together (dunno why I didn't do this earlier) and remove the `PhantomData`.